source/Files.cpp

```diff
@@ -58,70 +58,60 @@ namespace {
 				c = '/';
 	}
 #endif
-}
-
-
-
-void Files::Init(const char * const *argv)
-{
-	// Parse the command line arguments to see if the user has specified
-	// different directories to use.
-	for(const char * const *it = argv + 1; *it; ++it)
-	{
-		string arg = *it;
-		if((arg == "-r" || arg == "--resources") && *++it)
-			resources = *it;
-		else if((arg == "-c" || arg == "--config") && *++it)
-			config = *it;
-			
-	}
 	
-	if(resources.empty())
+	void SetResourceDirectory(string &resources)
 	{
 		// Find the path to the resource directory. This will depend on the
 		// operating system, and can be overridden by a command line argument.
 		char *str = SDL_GetBasePath();
 		if(!str)
 			throw runtime_error("Unable to get path to resource directory!");
-		
+
 		resources = str;
 		SDL_free(str);
-	}
+
+		// Create the "plugins" directory if it does not yet exist, so that it is
+		// clear to the user where plugins should go.
+		{
+			char *str = SDL_GetPrefPath("endless-sky", "plugins");
+			if(str != nullptr)
+				SDL_free(str);
+		}
+
 #if defined _WIN32
-	FixWindowsSlashes(resources);
+		FixWindowsSlashes(resources);
 #endif
-	if(resources.back() != '/')
-		resources += '/';
+		if(resources.back() != '/')
+			resources += '/';
```

Comment:
This is a pattern repeated a couple times (fixing slashes if windows and then ensuring the string ends in `/`), mind extracting to a helper function?